src/ast/semantic_analyser.cpp

```diff
@@ -1060,8 +1120,16 @@ void SemanticAnalyser::visit(Call &call)
   {
     if (!check_nargs(call, 1))
       return;
-    if (!check_arg(call, Type::pointer, 0))
+
+    // kptr should accept both integer or pointer. Consider case: kptr($1)
+    auto &arg = *call.vargs->at(0);
```

Comment:
What is the use case for this? 

Setting the addrspace should only be valid on pointers as pointers should be the only thing that we have to probe read. 

If we need `kptr($1)` to read data from a user specified  the resulting type should be `uint64 *` as per #1507, not just an int type.

Reply:
Ah i guess `kptr(arg0)` is a valid use case, in that case we should do the implicit cast e.g. `kptr(arg0)` --> `kptr((uint64*) arg0)`

Reply:
Ok. Cast looks better. I was thinking the following: If we know some kernel address and store that in a variable, then perform kptr($a). We would not need cast. But it it conflicts with the language, then we can surely change it.